server/src/test/java/org/elasticsearch/discovery/ClusterDisruptionIT.java

```diff
@@ -436,4 +442,54 @@ public void testIndicesDeleted() throws Exception {
         assertFalse(client().admin().indices().prepareExists(idxName).get().isExists());
     }
 
+    public void testRestartPrimaryNodeWhileIndexing() throws Exception {
```

Comment:
I will combine this newly added test to `testAckedIndexing` in a follow-up.

Reply:
was this test able to expose the bug without the fix?